examples/custom_repository.js

```diff
@@ -0,0 +1,34 @@
+const { EventEmitter } = require('events');
+const { initialize, isEnabled } = require('../lib');
+
+class MyRepo extends EventEmitter {
+    constructor() {
+        super();
+        this.data = {
+            name: 'my-feature',
+            description: 'foo',
+            enabled: true,
+            strategies: [],
+            variants: [],
+        };
+    }
+    stop() {
+        return;
+    }
+    getToggle() {
+        return this.data;
+    }
+}
+const repo = new MyRepo();
+const client = initialize({
+    appName: 'my-application',
+    url: 'not-needed',
+    disableMetrics: true,
+    repository: repo,
+});
+repo.emit('ready');
```

Comment:
this *really* bugs my eyes. TODO: add more docs about the requirements of the repo implementation

Reply:
Agreed. I think all events need to be properly documented. I also think there is a missing event today; `update` used to inform the client/user that there exist updated feature toggles (would be emitted when unleash-server returns 200 and not when it only receives 304).